### ISSUE TITLE
Display spoilers in text messages 

### ISSUE THREAD

**Issue by kevinaboos** [project-robius/robrix#274]
Currently Robrix does not display anything at all for text message content that is wrapped within a `<data-mx-spoiler>` tag. This is bad and makes it seem like the message is empty. 

The easiest way to implement this is to make the message unreadable by setting the text background color to be the same as the foreground color. 
(Ideally, we would use a fancy blur animation, but that's quite difficult to implement.) 

Then, when the user clicks the spoiler text, we should "reveal" the text and make it readable, e.g., by setting the background color of the text back to normal.

The *key* to this implementation is to show the `Cursor` hand upon hovering over the spoiler text, otherwise the user may now be aware that there's anything clickable there. 

With this implementation we can optionally choose to hide or show the spoiler reason (if one is given). That's not super important in the first step.

**Comment by rabelmervin** [comment]
Hi sir @kevinaboos Can i contribute to this issue ? This will be a good learning opportunity for me.

**Comment by kevinaboos** [comment]
> Hi sir @kevinaboos Can i contribute to this issue ? This will be a good learning opportunity for me.

definitely, go for it! We love to have contributions from anyone, they're much appreciated.

**Comment by rabelmervin** [comment]
Thanks @kevinaboos Excited to work on this !

**Comment by rabelmervin** [comment]
Hi @kevinaboos it would be great if you tell me the file location where changes needed.

**Comment by kevinaboos** [comment]
Well you'll need to modify a few different things to implement this. First up would be to create a Spoiler widget, probably in its own separate file, which includes the blurred or colored-over view of the hidden text and the spoiler reason text, both as Rust states. 
Then, you'll need to add this new Spoiler widget into the `RobrixHtml` widget (in `html_or_text.rs`) such that it is used when the `span` tag includes a `data-mx-spoiler` attribute. That's not super easy to do at the moment, but you can also take a look at Makepad's own Html widget to see how other custom widgets can be integrated into it. 

You'll also need to implement click handlers for the Spoiler widget such that it can transform info plaintext upon click.

**Comment by kevinaboos** [comment]
So this was a bit more complex than I initially thought, but I have just implemented both background colors and spoilers myself (they use similar techniques). @rabelmervin Just wanted to let you know in case you were still tinkering with it.

**Comment by rabelmervin** [comment]
Yeah @kevinaboos , Initially i struggled with this issue a lot Since, I was new to rust
